File: api/Models/Covenant.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Cretovale_api.Models
{
    public class Covenant : IEntity
    {
        public Guid Id { get; set; }

        [Required]
        public string Name { get; set; }

        public string Site { get; set; }
        //benefit
        public string Benefict { get; set; }
        public string ContactName { get; set; }
        public string ContactEmail { get; set; }
        public string Phone { get; set; }
        public string Status { get; set; }

        [Required]
        public Company Company { get; set; }

        public DateTime DateCreated { get; set; }
        public DateTime DateUpdated { get; set; }
    }
}

File: client/src/app/shared/models/covenant.ts
import { Company } from './covenant-company';

export class Covenant {
    constructor(public id?: number,
                public name?: string,
                public site?: string,
                public benefict?: string,
                public contactName?: string,
                public contactEmail?: string,
                public phone?: string,
                public status?: string,
                public covenantCategoryId?: number,
                public covenantCompanyId?: number,
                public company?: Company,
                public companyName?: string,
                public logo?: string) {
    }
}
